main.go

```diff
@@ -171,45 +171,12 @@ func fieldName(x interface{}) *dst.Ident {
 // fields of the struct based on specified flags
 func (c *config) modify(f *dst.File) (*dst.File, error) {
 	var foundOne bool
-	sortStructs := func(x dst.Node) bool {
-		var anon = []structType{}
-
-		// we need TypeSpec so as to
-		// parse the name of the struct
-		t, ok := x.(*dst.TypeSpec)
-		if !ok {
-			return true
-		}
-
-		if t.Type == nil {
-			return true
-		}
-
-		// if --struct is passed and no matches
-		// found, return appropriate response
-		if c.strct != "" && t.Name.Name == c.strct {
-			foundOne = true
-		}
-
-		// if this is the struct we want to modify
-		// get the StructType type and begin modification
-		s, ok := t.Type.(*dst.StructType)
-		if !ok {
-			return true
-		}
 
-		// now that the current node is indeed a struct
-		// if line number is provided, return if no match
-		if len(c.line) != 0 {
-			// convert dst node to ast and get position
-			startLNo := c.fset.Position(c.dec.Ast.Nodes[x].Pos()).Line
-			endLNo := c.fset.Position(c.dec.Ast.Nodes[x].End()).Line
-			if !(startLNo <= c.start && c.end <= endLNo) {
-				return true
-			}
-		}
+	var sortStructFields func(*dst.StructType)
+	sortStructFields = func(s *dst.StructType) {
```

Comment:
```suggestion
	sortStructFields := func(s *dst.StructType) {
```

unless it goes out scope when you recursively try to use it, shouldn't be the case though

Reply:
we can't do this, we won't be able to call the function on line 228